packages/devp2p/test/integration/util.ts

```diff
@@ -161,3 +161,48 @@ export function destroyRLPXs(rlpxs: any) {
     rlpx.destroy()
   }
 }
+
+export function twoPeerMsgExchange2(
+  t: Test,
+  opts: any,
+  capabilities?: any,
+  common?: Object | Common
+) {
+  const rlpxs = initTwoPeerRLPXSetup(null, capabilities, common)
+  rlpxs[0].on('peer:added', function (peer: any) {
+    const protocol = peer.getProtocols()[0]
+    const v4Hello = {
+      protocolVersion: 4,
+      clientId: 'fakePeer',
+      capabilities: [ETH.eth66],
+      port: 30303,
+      id: Buffer.alloc(12),
+    }
+    // Set peer's devp2p protocol version to 4
+    protocol._peer._hello = v4Hello
+    protocol.sendStatus(opts.status0)
+    peer.on('error', (err: Error) => {
+      t.fail(`Unexpected peer 0 error: ${err}`)
+    })
+  })
+
+  rlpxs[1].on('peer:added', function (peer: any) {
+    const protocol = peer.getProtocols()[0]
+    protocol.once('message', async (code: any, _payload: any) => {
+      switch (code) {
+        case ETH.MESSAGE_CODES.STATUS:
+          t.fail('should not have been able to process status message')
+          break
+      }
+    })
+    peer.once('error', (err: any) => {
+      t.equal(
+        err.message,
+        'Invalid Snappy bitstream',
+        'unable to process snappy compressed message'
+      )
+      t.end()
+      destroyRLPXs(rlpxs)
+    })
+  })
+}
```

Comment:
Hmm, this is somewhat a code doubling of a utility method meant to be somewhat generic more or less for one test. Not sure, can we have this a bit more generic, eventually also integrated in the first version of the function? Otherwise it's a bit misplaced here.

Didn't have "the solution" [tm] though as well on first look, just to admit.

Reply:
I don't know if it's possible to merge the tests from this second method into the `twoPeerMsgExchange` message since that isn't truly a utility method given there are specific test conditions that are checked within `twoPeerMsgExchange`.  We might be able to but seems to me it would make more sense to completely change how this set of tests is structures so that the specific test conditions are passed into the utility method (if that's even feasible).  Just my $0.02 though.

Reply:
Yeah, that's also a good point, just wanted to mention. The whole test structure of the devp2p tests is not so optimal yet. Would be happy if we at some point take the time to get to some substantial structural improvements here with both a better test setup - eventually also with some help of networking mocking test tools and/or libraries - as well as some more fine-grained unit tests on the different communication scenarios.

Well well. So much to do. 😋

Reply:
> Would be happy if we at some point take the time to get to some substantial structural improvements here with both a better test setup - eventually also with some help of networking mocking test tools and/or libraries - as well as some more fine-grained unit tests on the different communication scenarios.

yeah sounds great, maybe I can help with some of that, I would like to get more familiar with the devp2p internals and refactoring test code is always a good way to do that :)